src/main/java/ch/njol/skript/hooks/economy/expressions/ExprBalance.java

```diff
@@ -69,44 +66,37 @@ protected String getPropertyName() {
 	
 	@Override
 	@Nullable
-	public Class<?>[] acceptChange(final ChangeMode mode) {
+	public Class<?>[] acceptChange(ChangeMode mode) {
 		if (mode == ChangeMode.REMOVE_ALL)
 			return null;
 		return new Class[] {Money.class, Number.class};
 	}
 	
-	@SuppressWarnings("deprecation")
 	@Override
-	public void change(final Event e, final @Nullable Object[] delta, final ChangeMode mode) throws UnsupportedOperationException {
-		assert mode != ChangeMode.REMOVE_ALL;
-		
-		if (delta == null) {
-			for (final OfflinePlayer p : getExpr().getArray(e))
-				VaultHook.economy.withdrawPlayer(p.getName(), VaultHook.economy.getBalance(p.getName()));
+	public void change(Event e, @Nullable Object[] delta, ChangeMode mode) {
+		if (delta == null) { // RESET/DELETE
+			for (OfflinePlayer p : getExpr().getArray(e))
+				VaultHook.economy.withdrawPlayer(p, VaultHook.economy.getBalance(p));
 			return;
 		}
-		
-		final double m = delta[0] instanceof Number ? ((Number) delta[0]).doubleValue() : ((Money) delta[0]).getAmount();
-		for (final OfflinePlayer p : getExpr().getArray(e)) {
+
+		double m = delta[0] instanceof Number ? ((Number) delta[0]).doubleValue() : ((Money) delta[0]).getAmount();
+		for (OfflinePlayer p : getExpr().getArray(e)) {
 			switch (mode) {
 				case SET:
-					final double b = VaultHook.economy.getBalance(p.getName());
+					double b = VaultHook.economy.getBalance(p);
 					if (b < m) {
-						VaultHook.economy.depositPlayer(p.getName(), m - b);
+						VaultHook.economy.depositPlayer(p, m - b);
 					} else if (b > m) {
-						VaultHook.economy.withdrawPlayer(p.getName(), b - m);
+						VaultHook.economy.withdrawPlayer(p, b - m);
 					}
 					break;
 				case ADD:
-					VaultHook.economy.depositPlayer(p.getName(), m);
+					VaultHook.economy.depositPlayer(p, m);
 					break;
 				case REMOVE:
-					VaultHook.economy.withdrawPlayer(p.getName(), m);
+					VaultHook.economy.withdrawPlayer(p, m);
 					break;
-				case DELETE:
```

Comment:
I would actually just add:
```java
default:
	assert false;
```

Reply:
Is there a reason for that? we don't usually do that in other classes

Reply:
There are some places where it occurs, but I don't think it's a big deal for this PR :)